client/src/partials/journal/postingJournalController.js

```diff
@@ -0,0 +1,354 @@
+var PostingJournalController = function ($translate, $filter, $q, precision, sessionService, dataviewService, columnsService, gridService, dataLoaderService, liberror, messenger, Store, connect, $window, uuid, $rootScope, util, managerService) {
+  var vm = this;
+   //declaring a variable to handle error  
+  vm.managerService = managerService;
+  vm.dataviewService = dataviewService;
+  vm.columnsService  = columnsService;
+  vm.gridService = gridService;
+  vm.dataLoaderService = dataLoaderService;
+  vm.grid     = gridService.buildGrid();
+  vm.editing = false;
+  vm.project = sessionService.project;
+
+
+  vm.dataviewService.populate()
+  .then(vm.dataLoaderService.loadAdditionalData)  
+  .then(initialise)
+  .catch(handleErrors);  
+  
+  function isDefined (d) { return angular.isDefined(d); }
+  function handleErrors (error) {
+    messenger.danger('An error occured ' + JSON.stringify(error));
+  }
+
+  function initialise (models) {
+    angular.extend(vm, models);
+    vm.journal = new Store({ data : vm.dataviewService.dataview.getItems() });
+    vm.dataviewService.subscribeToOnRowCountChanged(vm.grid);
+    vm.dataviewService.subscribeToOnRowsChanged(vm.grid);
+    vm.gridService.subscribeToOnCellChange();
```

Comment:
Since you are calling four methods on gridService, could you group these into a single method call?  Something like
`gridService.initGridListeners()` or `gridService.initEventHandlers()`?